res/controllers/Hercules-P32-scripts.js

```diff
@@ -218,127 +229,164 @@ P32.Deck = function (deckNumbers, channel) {
+                    // way to set the loop size LED less than 1/32 (even though it
+                    // should be able to show 1/64).
                     engine.setValue(this.group, 'loop_halve', 1);
                     engine.setValue(this.group, 'loop_halve', 0);
-                } else if (value < 64 && loopSize < 64) { // turn right
-                    /**
-                        Mixxx supports loops longer than 64 beats, but the loop size LED
-                        only has 2 digits, so it couldn't show 128
-                    **/
-                    loopSize *= 2;
+                } else if (value < 64) { // turn right
+                    // Mixxx supports loops longer than 64 beats, but the loop size LED
+                    // only has 2 digits, so it couldn't show 128
                     engine.setValue(this.group, 'loop_double', 1);
                     engine.setValue(this.group, 'loop_double', 0);
                 }
             }
-            this.trigger();
         },
-        outKey: 'loop_enabled',
+        connect: function () {
+            this.connections[0] = engine.connectControl(this.group, 'beatloop_size', this.output);
+            if (loopEnabledDot) {
+                this.connections[1] = engine.connectControl(this.group, 'loop_enabled', this.output);
+            }
+        },
         output: function (value, group, control) {
-            if (loopEnabledDot && value) {
-                this.send(5 - Math.log(loopSize) / Math.log(2));
+            var loopSize = engine.getValue(this.group, 'beatloop_size');
+            var loopSizeLogBase2 = Math.log(loopSize) / Math.log(2);
+            // test if loopSizeLogBase2 is an integer
+            if (Math.floor(loopSizeLogBase2) === loopSizeLogBase2) {
+                if (loopEnabledDot && engine.getValue(this.group, 'loop_enabled') === 1) {
+                    this.send(5 - loopSizeLogBase2);
+                } else {
+                    this.send(5 + loopSizeLogBase2);
+                }
             } else {
-                this.send(5 + Math.log(loopSize) / Math.log(2));
+                this.send(14); // show two dots
             }
         }
     });
 
     this.loopMoveEncoder = function (channel, control, value, status, group) {
-        var direction = (value > 64) ? -1 : 1;
-        if (loopSize < 1) {
-            engine.setValue(this.currentDeck, 'loop_move', loopSize * direction);
-        } else {
-            engine.setValue(this.currentDeck, 'loop_move', 1 * direction);
+        if (value > 64) { // left turn
+            engine.setValue(this.currentDeck, 'beatjump_1_backward', 1);
+            engine.beginTimer(200, function () {
+                engine.setValue(this.currentDeck, 'beatjump_1_backward', 0);
+            }, true);
+        } else { // right turn
+            engine.setValue(this.currentDeck, 'beatjump_1_forward', 1);
+            engine.beginTimer(200, function () {
```

Comment:
But why delayed by 200ms? I trust the the code is necessary actually and I think I've had to do it as well - at least setting something to 1 and then back to 0 immediately without delay.

Would definitely be nice to have factored out, thanks!